src/main/java/nl/b3p/tailormap/api/persistence/helper/ApplicationHelper.java

```diff
@@ -306,9 +321,12 @@ private boolean addAppLayerItem(AppTreeLayerNode layerRef) {
 
       String legendImageUrl = serviceLayerSettings.getLegendImageId();
       if (legendImageUrl == null && serviceLayer.getStyles() != null) {
-        URI serviceLegendUrl =
-            serviceLayer.getStyles().stream().findFirst().map(WMSStyle::getLegendURL).orElse(null);
+        URI serviceLegendUrl = GeoServiceHelper.getLayerLegendUrlFromStyles(serviceLayer);
         legendImageUrl = serviceLegendUrl != null ? serviceLegendUrl.toString() : null;
+        if (null != legendImageUrl && proxied) {
```

Comment:
I might be wrong but if legendImageUrl == null && proxied, the frontend will do a GetLegendGraphic request to the (proxy) service URL which won't work, so maybe always set legendImageUrl if proxied is true?

Reply:
legendImageUrl is set to null when the service does not provide a legend url and and there is no user provided (image) url, if the frontend creates its own urls these may or may not work, but that is not the responsibility of the api... 

note that there is ia testcase for client-side generated GetLegendGraphic request: 
https://github.com/B3Partners/tailormap-api/blob/db3eaa53b9e1412cd4a70a0c72fecb0feaa4c091/src/test/java/nl/b3p/tailormap/api/controller/GeoServiceProxyControllerIntegrationTest.java#L206-L224